pkg/controller/drain.go

```diff
@@ -852,12 +852,13 @@ func (o *DrainOptions) evictPodWithoutPVInternal(attemptEvict bool, pod *corev1.
 	podArray := []*api.Pod{pod}
 
 	timeout := o.getTerminationGracePeriod(pod)
-	if timeout < o.Timeout {
-		glog.V(3).Infof("Overriding large termination grace period (%s) for the pod %s/%s", timeout.String(), pod.Namespace, pod.Name)
+	if timeout > o.Timeout {
```

Comment:
This check should also be there for pods with PV right?

In function `evictPodsWithPVInternal`, we have a main context with global timeout let's say 12 hours. After successful eviction/deletion, we have another local context with timeout = `o.getTerminationGracePeriod(pod)+o.PvDetachTimeout)` let's say 20 hours

But while waiting for PV detach we only check for the local context and not the global context. This makes the global timeout passed ineffective.

Reply:
[Global context](https://github.com/gardener/machine-controller-manager/blob/master/pkg/controller/drain.go#L600) should we passed in [this](https://github.com/gardener/machine-controller-manager/blob/master/pkg/controller/drain.go#L712) function and it should be checked [here](https://github.com/gardener/machine-controller-manager/blob/master/pkg/controller/drain.go#L724)

Reply:
[`Context.WithTimeout`](https://golang.org/pkg/context/#WithTimeout) is supposed to be the same as `WithDeadline(parent, time.Now().Add(timeout))` and [`Context.WithDeadline`](https://golang.org/pkg/context/#WithDeadline) is not supposed to override parent's deadline if the parent's deadline is earlier.

So, technically, no change required. But I see no harm in making it explicit either.

Reply:
ah ok. cool then.
/lgtm